src/file_storage.cpp

```diff
@@ -741,10 +712,13 @@ namespace {
 		e.hidden_attribute = bool(file_flags & file_storage::flag_hidden);
 		e.executable_attribute = bool(file_flags & file_storage::flag_executable);
 		e.symlink_attribute = bool(file_flags & file_storage::flag_symlink);
-		e.hash = filehash;
 		e.root = root_hash;
-		e.mtime = mtime;
 
+		if (filehash)
+		{
+			if (m_file_hashes.size() < m_files.size()) m_file_hashes.resize(m_files.size());
```

Comment:
The new elements in the vector need to be initialized to `nullptr`. Same goes for `m_time` below. And actually now that I look at it this issue is also in master/RC_1_2/RC_1_1.

Reply:
I'm pretty sure `std::vector` always default initialize elements, which will mean `0` and `nullptr` in this case

Reply:
and by "pretty sure" I really mean "confident". https://en.cppreference.com/w/cpp/container/vector/resize